cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -0,0 +1,219 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/data/table.hpp"
+#include "oneapi/dal/detail/common.hpp"
+
+namespace oneapi::dal::decision_forest {
+
+namespace detail {
+struct tag {};
+class descriptor_impl;
+class model_impl;
+} // namespace detail
+
+namespace task {
+struct classification {};
+struct regression {};
+using by_default = classification;
+} // namespace task
+
+namespace method {
+struct default_dense {};
+struct hist {};
+using by_default = default_dense;
+} // namespace method
+
+enum class variable_importance_mode
+{
+    none,      /* Do not compute */
+    mdi,       /* Mean Decrease Impurity.
+                       Computed as the sum of weighted impurity decreases for all nodes where the variable is used,
+                       averaged over all trees in the forest */
+    mda_raw,   /* Mean Decrease Accuracy (permutation importance).
+                       For each tree, the prediction error on the out-of-bag portion of the data is computed
+                       (error rate for classification, MSE for regression).
+                       The same is done after permuting each predictor variable.
+                       The difference between the two are then averaged over all trees. */
+    mda_scaled /* Mean Decrease Accuracy (permutation importance).
+                       This is MDA_Raw value scaled by its standard deviation. */
+};
+
+enum class result_to_compute_id
+{
+    compute_out_of_bag_error                 = 0x00000001ULL,
+    compute_out_of_bag_error_per_observation = 0x00000002ULL
+};
+
+enum class voting_method
+{
+    weighted = 0,
+    unweighted,
+    last_result_id = unweighted
+};
+
+class descriptor_base : public base {
+  public:
+    using tag_t = detail::tag;
+    using float_t = float;
+    using task_t   = task::by_default;
+    using method_t = method::by_default;
+
+    descriptor_base();
+
+    //engines::EnginePtr engine;             /*!< Engine for the random numbers generator used by the algorithms */
+
+    double get_observations_per_tree_fraction() const;
+    double get_impurity_threshold() const;
+    double get_min_weight_fraction_in_leaf_node() const;
+    double get_min_impurity_decrease_in_split_node() const;
+
+    std::int64_t get_n_classes() const;
+    std::int64_t get_n_trees() const;
+    std::int64_t get_features_per_node() const;
+    std::int64_t get_max_tree_depth() const;
+    std::int64_t get_min_observations_in_leaf_node() const;
+    std::int64_t get_seed() const;
+    std::int64_t get_min_observations_in_split_node() const;
+    std::int64_t get_max_leaf_nodes() const;
+
+    bool get_memory_saving_mode() const;
+    bool get_bootstrap() const;
+
+    std::uint64_t get_results_to_compute() const;
```

Comment:
Why it's `uint64_t`? Should be enum type.

Reply:
Due to it will be a combination (by |) of results